implementations/c/include/ockam/vault.h

```diff
@@ -76,88 +68,142 @@ typedef enum {
+ ********************************************************************************************************
+ */
 
-OCKAM_ERR ockam_vault_hkdf(uint8_t *p_salt, uint32_t salt_size,
-                           uint8_t *p_ikm, uint32_t ikm_size,
-                           uint8_t *p_info, uint32_t info_size,
-                           uint8_t *p_out, uint32_t out_size);
+OckamError VaultBearsslCreate(OckamVaultCtx **ctx,
+                              OckamVaultBearsslConfig *p_cfg,
+                              const OckamMemory *memory);
+
+OckamError VaultBearsslDestroy(OckamVaultCtx *ctx);
+
+OckamError VaultBearsslRandom(OckamVaultCtx *ctx,
+                              uint8_t *p_num, size_t num_size);
+
+OckamError VaultBearsslKeyGenerate(OckamVaultCtx *ctx,
+                                   OckamVaultKey key_type);
+
+OckamError VaultBearsslKeyGetPublic(OckamVaultCtx *ctx,
+                                    OckamVaultKey key_type,
+                                    uint8_t *p_pub_key, size_t pub_key_size);
+
+OckamError VaultBearsslKeySetPrivate(OckamVaultCtx *ctx,
+                                     OckamVaultKey key_type,
+                                     uint8_t *p_priv_key, size_t priv_key_size);
+
+OckamError VaultBearsslEcdh(OckamVaultCtx *ctx,
+                            OckamVaultKey key_type,
+                            uint8_t *p_pub_key, size_t pub_key_size,
+                            uint8_t *p_ss, size_t ss_size);
+
+OckamError VaultBearsslSha256(OckamVaultCtx *ctx,
+                              uint8_t *p_msg, size_t msg_size,
+                              uint8_t *p_digest, size_t digest_size);
+
+OckamError VaultBearsslHkdf(OckamVaultCtx *ctx,
+                            uint8_t *p_salt, size_t salt_size,
+                            uint8_t *p_ikm, size_t ikm_size,
+                            uint8_t *p_info, size_t info_size,
+                            uint8_t *p_out, size_t out_size);
+
+OckamError VaultBearsslAesGcmEncrypt(OckamVaultCtx *ctx,
+                                     uint8_t *p_key, size_t key_size,
+                                     uint8_t *p_iv, size_t iv_size,
+                                     uint8_t *p_aad, size_t aad_size,
+                                     uint8_t *p_tag, size_t tag_size,
+                                     uint8_t *p_input, size_t input_size,
+                                     uint8_t *p_output, size_t output_size);
+
+OckamError VaultBearsslAesGcmDecrypt(OckamVaultCtx *ctx,
+                                     uint8_t *p_key, size_t key_size,
+                                     uint8_t *p_iv, size_t iv_size,
+                                     uint8_t *p_aad, size_t aad_size,
+                                     uint8_t *p_tag, size_t tag_size,
+                                     uint8_t *p_input, size_t input_size,
+                                     uint8_t *p_output, size_t output_size);
```

Comment:
rename `bearssl` vault to `default` and move these functions to that impl header.